# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: D:\CM-work\hudson\plugins\versionnumber
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberAction.java
--- src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberAction.java Base (BASE)
+++ src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberAction.java Locally Modified (Based On LOCAL)
@@ -1,16 +1,21 @@
 package org.jvnet.hudson.tools.versionnumber;
 
+import hudson.EnvVars;
+import hudson.model.AbstractBuild;
 import hudson.model.Action;
+import hudson.model.EnvironmentContributingAction;
 
-public class VersionNumberAction implements Action {
+public class VersionNumberAction implements Action, EnvironmentContributingAction {
 	static final String ICON = "/plugin/versionnumber/vnicon_24x24.gif";
 	
 	private VersionNumberBuildInfo info;
 	private String versionNumber;
+	private String environmentVariableName;
 
-	public VersionNumberAction(VersionNumberBuildInfo info, String versionNumber) {
+	public VersionNumberAction(VersionNumberBuildInfo info, String versionNumber, String environmentVariableName) {
 		this.info = info;
 		this.versionNumber = versionNumber;
+		this.environmentVariableName = environmentVariableName;
 	}
 	
 	public VersionNumberBuildInfo getInfo() {
@@ -29,5 +34,7 @@
 		return "versionnumber/displayName";
 	}
 
-	
+	public void buildEnvVars(AbstractBuild<?, ?> build, EnvVars env) {
+		env.put(this.environmentVariableName, this.versionNumber);
 }
+}
Index: src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java
--- src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java Base (BASE)
+++ src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java Locally Modified (Based On LOCAL)
@@ -348,7 +348,7 @@
 					build.getTimestamp(),
 					listener.getLogger()
 					);
-    		build.addAction(new VersionNumberAction(info, formattedVersionNumber));
+    		build.addAction(new VersionNumberAction(info, formattedVersionNumber, this.environmentVariableName));
 		} catch (IOException e) {
 			// TODO Auto-generated catch block
 			listener.error(e.toString());
@@ -361,14 +361,8 @@
 			listener.error(e.toString());
 			build.setResult(Result.FAILURE);
 		}
-		final String finalVersionNumber = formattedVersionNumber;
-        return new Environment() {
-        	@Override
-        	public void buildEnvVars(Map<String, String> env) {
-        		env.put(environmentVariableName, finalVersionNumber);
+        return new Environment() {};
         	}
-        };
-    }
 
     @Override
     public BuildWrapperDescriptor getDescriptor() {
